src/universal/modules/userDashboard/components/OrgAvatarInput/OrgAvatarInput.js

```diff
@@ -16,18 +16,19 @@ const validate = (values) => {
 };
 
 const uploadPicture = async (atmosphere, orgId, pictureFile) => {
-  return new Promise((resolve, reject) => {
-    const variables = {
-      contentType: pictureFile.type,
-      contentLength: pictureFile.size,
-      orgId
-    };
-    const onError = (err) => {
-      reject(err);
-    };
-    const onCompleted = (res) => {
+  const variables = {
+    contentType: pictureFile.type,
+    contentLength: pictureFile.size,
+    orgId
+  };
+  const onError = (err) => {
+    throw new Error(err);
+  };
+
+  return new Promise((resolve) => {
+    const onCompleted = async (res) => {
       const {createOrgPicturePutUrl: {url}} = res;
-      const pathname = sendAssetToS3(pictureFile, url);
+      const pathname = await sendAssetToS3(pictureFile, url);
```

Comment:
if AWS goes down, we've got much bigger problems than a profile picture :wink:

this is a good point though, we should talk about when to catch `await` clauses. for example, on the server, we always assume our DB is up. if a mutation triggers 2 writes & the 2nd write fails, we don't roll back the first (the exception being around payment stuff. i make sure to avoid race conditions there).

IMO, the effort still isn't worth the reward. if AWS fails, then they'll hit refresh & try again nbd.

Reply:
> if AWS goes down, we've got much bigger problems than a profile picture 😉

This could fail for a number of reasons, of which AWS "going down" is just one!

> IMO, the effort still isn't worth the reward. if AWS fails, then they'll hit refresh & try again nbd.

I mean, I hear ya, but I was taught to always catch errors. Even if we just pop a toast saying "try again"...